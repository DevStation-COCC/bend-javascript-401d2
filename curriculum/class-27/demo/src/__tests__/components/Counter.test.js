import React from 'react';
import renderer from 'react-test-renderer';
import Counter from '../../components/Counter';

describe('<Counter/> Enzyme Test', () => {
  it('is alive at application start', () => {
    // eslint-disable-next-line no-undef
    const app = shallow(<Counter/>);

    expect(app.find('.count').text()).toBe('0');
  });

  it('can count up', () => {
    // eslint-disable-next-line no-undef
    const wrapper = mount(<Counter/>);

    wrapper.find('.increment').simulate('click');
    expect(wrapper.state('counter')).toBe(1);
    wrapper.find('.increment').simulate('click');
    expect(wrapper.state('counter')).toBe(2);
  });

  it('can count down', () => {
    // eslint-disable-next-line no-undef
    const wrapper = mount(<Counter/>);

    wrapper.find('.decrement').simulate('click');
    expect(wrapper.state('counter')).toBe(-1);
    wrapper.find('.decrement').simulate('click');
    expect(wrapper.state('counter')).toBe(-2);
  });
});

describe('<Counter/> Snapshot Test', () => {
  it('renders correctly', () => {
    const component = renderer.create(<Counter/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
