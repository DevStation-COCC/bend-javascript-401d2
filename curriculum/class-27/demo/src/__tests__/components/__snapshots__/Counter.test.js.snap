// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter/> Snapshot Test renders correctly 1`] = `
<div
  className="counter"
>
  <h4
    className="count"
  >
    0
  </h4>
  <span
    className="decrement"
    onClick={[Function]}
  >
    Decrement
  </span>
  <span
    className="increment"
    onClick={[Function]}
  >
    Increment
  </span>
</div>
`;
